File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.swift
import Foundation

func solution(_ video_len:String, _ pos:String, _ op_start:String, _ op_end:String, _ commands:[String]) -> String {
    var answer = pos
    for c in commands {
        answer = skip(answer, op_start, op_end)
        if c == "next" {
            answer = next(video_len, answer)
        } else {
            answer = prev(video_len, answer)
        }
        answer = skip(answer, op_start, op_end)
        
    }
    return answer
}

func next(_ video_len:String, _ pos: String) -> String {
    let videoTime:[Int] = video_len.split { $0 == ":" }.map { Int($0)! }
    let (vmin, vsec) = (videoTime[0], videoTime[1])
    let vtotal = vmin*60 + vsec
    
    let times:[Int] = pos.split { $0 == ":" }.map { Int($0)! }
    let (min, sec) = (times[0], times[1])
    var total = min*60 + sec
    
    if vtotal - total < 10 {
        return video_len
    } else {
       total += 10
        let tempMin = String(total/60)
        let tempSec = String(total%60)
        
        var rMin = tempMin.count == 1 ? "0"+tempMin : tempMin
        var rSec = tempSec.count == 1 ? "0"+tempSec : tempSec
        
        return rMin+":"+rSec
    }
}

func prev(_ video_len:String, _ pos: String) -> String {
    let videoTime:[Int] = video_len.split { $0 == ":" }.map { Int($0)! }
    let (vmin, vsec) = (videoTime[0], videoTime[1])
    let vtotal = vmin*60 + vsec
    
    let times:[Int] = pos.split { $0 == ":" }.map { Int($0)! }
    let (min, sec) = (times[0], times[1])
    var total = min*60 + sec
    
    if total < 10 {
        return "00:00"
    } else {
       total -= 10
        let tempMin = String(total/60)
        let tempSec = String(total%60)
        
        var rMin = tempMin.count == 1 ? "0"+tempMin : tempMin
        var rSec = tempSec.count == 1 ? "0"+tempSec : tempSec
        
        return rMin+":"+rSec
    }
    
}

func skip(_ pos: String,_ op_start:String, _ op_end:String) -> String {
    if op_start <= pos && pos <= op_end {
        return op_end
    } else {
        return pos
    }
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def getss(time):
    times = list(map(int,time.split(":"))) 
    return times[0] * 60 + times[1]

def getms(time):
    mm = time//60
    ss = time%60
    cm = ""
    cs = ""
    
    if mm < 10:
        cm = "0"+str(mm)
    else:
        cm = str(mm)
    
    if ss == 0:
        cs = "00"
    elif ss < 10:
        cs = "0"+str(ss)
    else:
        cs = str(ss)
        
    return cm + ":" + cs

def c_next(time, video_len):
    video = getss(video_len)
    next = getss(time) + 10
    return getms(video) if video - next < 10 else getms(next)
    
def c_prev(time):
    prev = getss(time) - 10
    return getms(prev) if prev > 10 else "00:00"

def is_opening(time, s, e):
    now = getss(time)
    op_s = getss(s)
    op_e = getss(e)
    if op_s <= now <= op_e:
        return e
    else:
        return time
    
    

def solution(video_len, pos, op_start, op_end, commands):
    answer = pos
    for c in commands:
        answer = is_opening(answer, op_start, op_end)
        if c == "next":
            answer = c_next(answer, video_len)
        else:
            answer = c_prev(answer)
        answer = is_opening(answer, op_start, op_end)
    
    return answer